src/main/java/org/jabref/gui/mergeentries/MultiEntryMergeWithFetchedDataAction.java

```diff
@@ -0,0 +1,185 @@
+
+package org.jabref.gui.mergeentries;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import javafx.application.Platform;
+
+import org.jabref.gui.LibraryTab;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.ActionHelper;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.preferences.GuiPreferences;
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.logic.importer.fetcher.MergingIdBasedFetcher;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.BackgroundTask;
+import org.jabref.logic.util.NotificationService;
+import org.jabref.logic.util.TaskExecutor;
+import org.jabref.model.entry.BibEntry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles the merging of multiple bibliography entries with fetched data.
+ * This action performs background fetching and merging of entries while
+ * providing progress updates to the user.
+ */
+public class MultiEntryMergeWithFetchedDataAction extends SimpleCommand {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MultiEntryMergeWithFetchedDataAction.class);
+
+    private final Supplier<LibraryTab> tabSupplier;
+    private final GuiPreferences preferences;
+    private final NotificationService notificationService;
+    private final TaskExecutor taskExecutor;
+
+    public MultiEntryMergeWithFetchedDataAction(Supplier<LibraryTab> tabSupplier,
+                                                GuiPreferences preferences,
+                                                NotificationService notificationService,
+                                                StateManager stateManager,
+                                                TaskExecutor taskExecutor) {
+        this.tabSupplier = tabSupplier;
+        this.preferences = preferences;
+        this.notificationService = notificationService;
+        this.taskExecutor = taskExecutor;
+        this.executable.bind(ActionHelper.needsDatabase(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        LibraryTab libraryTab = tabSupplier.get();
+        if (libraryTab == null) {
+            LOGGER.error("Action 'Multi Entry Merge' must be disabled when no database is open.");
+            return;
+        }
+
+        List<BibEntry> entries = libraryTab.getDatabase().getEntries();
+        if (entries.isEmpty()) {
+            notificationService.notify(Localization.lang("No entries exist."));
```

Comment:
Re-use existing string `empty library`- we do not want to have different terms for the same thing.